iocore/eventsystem/ProtectedQueue.cc

```diff
@@ -56,55 +56,7 @@ ProtectedQueue::enqueue(Event *e, bool fast_signal)
     // queue e->ethread in the list of threads to be signalled
     // inserting_thread == 0 means it is not a regular EThread
     if (inserting_thread != e_ethread) {
-      if (!inserting_thread || !inserting_thread->ethreads_to_be_signalled) {
-        signal();
-        if (fast_signal) {
-          if (e_ethread->signal_hook) {
-            e_ethread->signal_hook(e_ethread);
-          }
-        }
-      } else {
-#ifdef EAGER_SIGNALLING
-        // Try to signal now and avoid deferred posting.
-        if (e_ethread->EventQueueExternal.try_signal())
-          return;
-#endif
-        if (fast_signal) {
-          if (e_ethread->signal_hook) {
-            e_ethread->signal_hook(e_ethread);
-          }
-        }
-        int &t          = inserting_thread->n_ethreads_to_be_signalled;
-        EThread **sig_e = inserting_thread->ethreads_to_be_signalled;
-        if ((t + 1) >= eventProcessor.n_ethreads) {
-          // we have run out of room
-          if ((t + 1) == eventProcessor.n_ethreads) {
-            // convert to direct map, put each ethread (sig_e[i]) into
-            // the direct map loation: sig_e[sig_e[i]->id]
-            for (int i = 0; i < t; i++) {
-              EThread *cur = sig_e[i]; // put this ethread
-              while (cur) {
-                EThread *next = sig_e[cur->id]; // into this location
-                if (next == cur) {
-                  break;
-                }
-                sig_e[cur->id] = cur;
-                cur            = next;
-              }
-              // if not overwritten
-              if (sig_e[i] && sig_e[i]->id != i) {
-                sig_e[i] = nullptr;
-              }
-            }
-            t++;
-          }
-          // we have a direct map, insert this EThread
-          sig_e[e_ethread->id] = e_ethread;
-        } else {
-          // insert into vector
-          sig_e[t++] = e_ethread;
```

Comment:
The `n_ethreads_to_be_signalled` is increased here but it is removed. It makes the `n_ethreads_to_be_signalled` is always 0.